src/chainrules.jl

```diff
@@ -0,0 +1,11 @@
+_NamedDimsArray_pullback(ȳ::AbstractArray) = (NoTangent(), ȳ, NoTangent())
```

Comment:
I am too slow but shouldn't this remove names from the array it gets? On the assumption that similar or broadcasting in the computation of a gradient for the array with names may, probably should, produce another array with names?

```
julia> nda = NamedDimsArray(rand(Int8,3,4), (:x, :y));

julia> gradient(x -> sum(x), nda)  # does not know about names
(Fill(1, 3, 4),)

julia> gradient(x -> sum(x; dims=1)[1], nda)  # involves broadcasting, so might? 
ERROR: DimensionMismatch("destination axes (Base.OneTo(3), Base.OneTo(4)) are not compatible with source axes (Base.OneTo(1), Base.OneTo(4))")

julia> gradient(x -> sum(x; dims=1)[1], nda.data)
(Int8[1 0 0 0; 1 0 0 0; 1 0 0 0],)
```

Reply:
thanks, I have opened an issue